### PR TITLE
🏷 Null safety support

### PR DESCRIPTION
**Pull request by lig**
The code has been migrated to null safety.
However, `flutter_colorpicker` has not released the null safety compatible version yet. However, it has null safety support in the main branch. Thus we need to use a dependency override for now.

### CODE REVIEW

**Comment by lig** [comment]
It's better to merge this one into a separate branch. See #24.

File: lib/property_widgets/border_radius_property.dart [hunk 1]
```diff
@@ -109,7 +110,9 @@ class _BorderRadiusPropertyState extends State<BorderRadiusProperty> {
             onPressed: () async {
               await show();
               widget.property.value = _currentBorderRadius;
-              widget.onChanged(widget.property);
+              if (widget.onChanged != null) {
```
Comment (erickzanardo) [inline lib/property_widgets/border_radius_property.dart:113]: we can use Elvis operator here instead right? `widget.onChange?.call(widget.property)`

File: lib/property_widgets/color_property.dart [hunk 2]
```diff
@@ -66,7 +65,9 @@ class ColorPropertyState extends State<ColorProperty> {
         onPressed: () async {
           await show();
           widget.property.value = currentColor;
-          widget.onChanged(widget.property);
+          if (widget.onChanged != null) {
```
Comment (erickzanardo) [inline lib/property_widgets/color_property.dart:68]: Here, too, lets use the elvis operator

File: lib/property_widgets/edge_insets_property.dart [hunk 3]
```diff
@@ -102,7 +103,9 @@ class _EdgeInsetsPropertyState extends State<EdgeInsetsProperty> {
             onPressed: () async {
               await show();
               widget.property.value = _currentEdgeinsets;
-              widget.onChanged(widget.property);
+              if (widget.onChanged != null) {
```
Comment (erickzanardo) [inline lib/property_widgets/edge_insets_property.dart:106]: Ditto

File: lib/property_widgets/list_property.dart [hunk 4]
```diff
@@ -25,7 +25,9 @@ class ListPropertyState extends State<ListPropertyWidget> {
         value: widget.property.getValue(),
         onChanged: (value) {
           widget.property.value = value;
-          widget.onChanged(widget.property);
+          if (widget.onChanged != null) {
```
Comment (erickzanardo) [inline lib/property_widgets/list_property.dart:28]: Ditto

File: lib/property_widgets/number_property.dart [hunk 5]
```diff
@@ -34,7 +34,9 @@ class NumberPropertyState extends State<NumberProperty> {
         inputFormatters: [FilteringTextInputFormatter.digitsOnly],
         onChanged: (value) {
           widget.property.value = double.tryParse(value);
-          widget.onChanged(widget.property);
+          if (widget.onChanged != null) {
```
Comment (erickzanardo) [inline lib/property_widgets/number_property.dart:37]: Ditto

File: lib/property_widgets/text_property.dart [hunk 6]
```diff
@@ -30,7 +30,9 @@ class TextPropertyState extends State<TextProperty> {
       child: TextField(
         onChanged: (value) {
           widget.property.value = value;
-          widget.onChanged(widget.property);
+          if (widget.onChanged != null) {
```
Comment (erickzanardo) [inline lib/property_widgets/text_property.dart:33]: Ditto

File: lib/story.dart [hunk 7]
```diff
@@ -137,16 +102,14 @@ class Chapter {
...
-  }
+  Chapter(this.name, this._buildFn, this.story) : ctx = DashbookContext();
 
   Widget widget(BoxConstraints constraints) {
     ctx.constraints = constraints;
     final Widget w = _buildFn(ctx);
 
     if (story._decorator != null) {
-      return story._decorator.decorate(w);
+      return story._decorator!.decorate(w);
```
Comment (erickzanardo) [inline lib/story.dart:112]: I know this if was not written by you, but lets change it to use the elvis operator as well

**Review by erickzanardo — COMMENTED** [review]
Some comments, but overall, changes looking good

File: lib/story.dart (hunk 7 above)
Comment (lig) [reply lib/story.dart:112]: Sure. It looks like I don't see elvis possible for a void method call. Interesting observation about myself:)

**Comment by lig** [comment]
> Some comments, but overall, changes looking good

Thanks! I've updated the PR.

**Comment by lig** [comment]
I'd recommend creating a new branch, i.e. `null-safety` to keep this code there until Dart 2.12 comes to the stable channel.

**Comment by lig** [comment]
@erickzanardo It's ready. Please, create a separate branch, i.e. `null-safety`, for me to retarget this PR to it.

This code is ready to be used as a dependency override. Thus, it is ok to have it in a separate branch.

However, it is not ready to be released even as a pre-release package yet as it contains overridden `flutter_colorpicker` dependency. See https://github.com/mchome/flutter_colorpicker/issues/35#issuecomment-738577044

**Comment by erickzanardo** [comment]
> null-safety

Sorry for the delay, I created the branch `null-safety`

**Comment by lig** [comment]
@erickzanardo thanks! I've retargeted the PR.

**Comment by erickzanardo** [comment]
> @erickzanardo thanks! I've retargeted the PR.

cool, I think I am going to merge this on the branch, but we will probably need a follow up PR when the color picker dependency is released right?

**Comment by lig** [comment]
@erickzanardo exactly. However, merging this will allow using the branch in dependency override for `dashbook` in other projects that depend on `dashbook`.

**Comment by erickzanardo** [comment]
> @erickzanardo exactly. However, merging this will allow using the branch in dependency override for `dashbook` in other projects that depend on `dashbook`.

Cool, merging it now! Thanks for this

**Pull request merged**